### DISCUSSION TITLE
Not deleting persistant data when stack is deleted

### DISCUSSION THREAD

**Discussion by isvein** [louislam/dockge#671] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

When I delete an stack, all the persistent data also gets deleted.

### ✔️ Solution

I would like to be able to choose if persistent data should be deleted or not so I dont loose important data if I accidentally deletes and stack and/or forgot to take an backup.

**Comment by mizady** [comment]
Are you creating the volumes with the stack or are you creating the volumes outside the stack and referencing them in the stack?  if you're using named volumes that the stack is creating, then this is actually likely correct behavior as removing/deleting a stack should delete anything that it created as well.

**Reply by isvein** [reply]
Im creating the volumes with the stack.
Or its just mappings, to location on the disk, but the stack create all the folders. My mappings are like "./dockage/stacks/stack-name"

When a stack is deleted, the "stack-name" folder gets deleted.

**Reply by mizady** [reply]
ah yes so that's because deleting the stack in dockge would be the stack, which is everything in the folder.  so really we need a pop up that has a check box on whether to delete the folder as well.  I'll see if i can throw something together in the next day or two, but keep in mind that the author of this project is busy and has not been approving very many pull requests so even if I throw the code up in a pull request it maybe some time before it makes it live.

**Reply by isvein** [reply]
Thanks :-) 
Cant ask for more.
I feel this will benefit more than just me.

**Reply by Celer21** [reply]
I had the same request in this post https://github.com/louislam/dockge/discussions/405 .
It coud be better to have check box to let user deside to delete only compose.yml and .env files or whole stack folder with data.

Something like this:
![image](https://github.com/user-attachments/assets/accaf251-40a7-4578-a0d8-7755a263d8bd)

**Reply by rasulovk** [reply]
Hi, any updates regarding this issue?
